.github/workflows/python-build-and-upload-packages.yml

```diff
@@ -0,0 +1,63 @@
+name: Python - build and upload packages
+
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'python-v*'
+  pull_request:
+    # In theory we wouldn't need to create artifacts for each PR, but this makes
+    # it much easier to test this workflow and there are no big cons.
+    paths:
+      - 'python/**'
+      - 'rust/**'
+      - 'tests_data/**'
+      - '.github/workflows/**'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: ${{ matrix.conf.name }}
+    runs-on: ${{ matrix.conf.runner }}
+    strategy:
+      matrix:
+        conf:
+          - name: linux
+            runner: ubuntu-latest
+          - name: windows
+            runner: windows-latest
+          - name: macos
+            runner: macos-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # pin@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # pin@v5
+        with:
+          python-version-file: 'python/.python-version'
+
+      - name: Install uv
+        run: curl -LsSf https://astral.sh/uv/0.4.7/install.sh | sh
+
+      # It's OK to use the default python version and not the specified one
+      - name: Install all projects dependencies
+        working-directory: python
+        run: uv sync --all-extras --dev
+
+      - name: Build the python package
+        working-directory: python
+        run: uv run ./scripts/build_python_package.py
+
+      - name: List wheels
+        working-directory: python
+        run: python -c "import os; print('\n'.join(os.listdir('dist')))"
```

Comment:
It seems like we create the source wheel for all runners, thus using 3 times more space as needed. Is there a way to use `uv build` to only build either the source or binary package? We could then have 4 configs in the matrix: linux, macos, windows, source. And build 4 artifacts that we can unzip in the same directory without overwriting a tar-ball with a similar tar-ball.

Reply:
true. instead of making the build python package script more complicated, for now I modified the workflow to upload the tarball only for linux. This has also the advantage of having 3 runners instead of 4. it's not ideal on some aspects, I propose to fix later if needed.

Reply:
That's also a good solution. I don't think we need to improve it.